python/flex.py

```diff
@@ -46,22 +53,32 @@ def __initRadio(self):
         # Here we wait until we have a connected radio
         while FlexApi.radio is None:
             pass
-        
+
         # Remove our event handler
         API.RadioAdded -= self.__radio_added
 
         self.__print_radio_info(FlexApi.radio)
         print("Connecting...")
         FlexApi.radio.Connect()
 
+        # Collect active panadapters and destroy them all (even the younglings)
+        pan_check = FlexApi.radio.WaitForPanadaptersSync()
+        print("flex::WaitForPanadaptersSync")
+        if len(pan_check) is not 0:
+            print("There are active panadapters...")
+            destroyed_pans = 0
+            print("Destroying panadapters...")
+            while len(pan_check) is not 0:
+                for p in pan_check:
+                    p.Close(True)
+                    destroyed_pans += 1
+                pan_check = FlexApi.radio.WaitForPanadaptersSync()
+        try:
+            print("{0} panadapters destroyed".format((destroyed_pans - 1) / 2))
```

Comment:
Do we need this print statement in a try catch?

Reply:
Ah I see what's happened here.  It's because `destroyed_pans` may not be defined at this point

Reply:
Ok, I wonder if we could make things a little bit more succinct:

```python
        print("flex::WaitForPanadaptersSync")  # Log what we do before, since the syncrhonous call below will block
        existing_pans = FlexApi.radio.WaitForPanadaptersSync()

        if len(existing_pans) == 0:
           print("No panadapters active.")
        else:
             print("Destroying panadapters...")
             destroyed_pans = 0
             while len(existing_pans) is not 0:
                 for p in existing_pans:
                     p.Close(True)
                     destroyed_pans += 1
                 existing_pans = FlexApi.radio.WaitForPanadaptersSync()
             print("{0} panadapters destroyed".format((destroyed_pans - 1) / 2))
```

Notice that you no longer need the `try/catch` because destroyed_pans is always defined in that scope.  Looking at it, do you still need to do the loop?  Could you use the new parameter that we added into the WaitForPanadaptersSync? i.e. `FlexApi.radio.WaitForPanadaptersSync(2000,1000) # wait a 1 second before returning panadapters list`